### PR TITLE
Edit User header isn't updated when user's name is changed

### PR DESCRIPTION
**Pull request by lukenewman**
## Background
* Set the user to be the newly-saved user.
* Compile fixes (component library fixed to 1.1.21 and added a type def to `AnnualHoseholdIncome.tsx`

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1404

## Associated PRs
None

## Validation Plan
Followed the repro steps in the issue description and validated the expected behavior.

## Automated Testing
- [ ] All unit tests are passing.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@ctoec/component-library": "^1.1.21",
+    "@ctoec/component-library": "1.1.21",
```
Comment (vonbearshark) [inline client/package.json:6]: Why this change?

Comment (vonbearshark) [reply client/package.json:6]: Also there's lots of lockfile diffs looks like?

**Review by vonbearshark — COMMENTED** [review]
Seems like the actual code change is small. Lots of `prettier` noise, it looks like. Not your fault and happened to others as well. We should investigate at some point. But as long as this style seems reasonable enough then this particular PR is fine there imo.

Would approve but waiting on confirmation for the package/lockfile changes

File: client/package.json (hunk 1 above)
Comment (lukenewman) [reply client/package.json:6]: for context: https://skylight-hq.slack.com/archives/CM7D99AF3/p1628717533010200

Comment (JordanGuinn) [reply client/package.json:6]: Was a `yarn install` run here by chance?  Instead of a `yarn install --frozen-lockfile`?

**Review by bamader — APPROVED** [review]
Content-wise this does what it's supposed to. I don't know wtf is going on with `AnnualHouseholdIncome` because I swear I've made that exact change in like seven different PRs already.

**Review by JordanGuinn — APPROVED** [review]

**Comment by bamader** [comment]
## Validation Plan
- Log into app with an admin user
- Go to edit user page for some dummy user account
- [ ] Can change user's name, and new name updates to what's on the page title
- [ ] Can add/remove orgs and change site perms as normal

**Review by vonbearshark — APPROVED** [review]

**Comment by bamader** [comment]
For whatever reason github isn't letting me mark the validation boxes as checked, but this passes all validation tests and is ready to go!

File: client/package.json (hunk 1 above)
Comment (JordanGuinn) [reply client/package.json:6]: Updated the lockfile to match the results of `yarn install --frozen-lockfile`
